### ISSUE TITLE
Maya Instanced geometry support in ```model``` subset / Validator

### ISSUE THREAD

**Issue by LiborBatek** [pypeclub/OpenPype#3991]
When trying to publish ```model``` family I get validation errors because of non-unique IDs of the geometries which shares its ```ShapeNode``` due to instance nature of them using Maya ```DuplicateSpecial``` function with geometry type se to ```Instance```

![image](https://user-images.githubusercontent.com/112623825/196110651-4a034fdb-31e8-4b70-b8b2-91ec5d393c2d.png)

Instancing in maya being crucial function and should be supported at least on a model level (look should be applied normally just multiple times I guess because of the same ```shapeNode``` and object ID...

**Comment by BigRoy** [comment]
> Instancing in maya being crucial function and should be supported at least on a model level (look should be applied normally just multiple times I guess because of the same shapeNode and object ID...

It's actually an unfortunate downside of the shaders being based on the `cbId`. In Maya you can have different shaders per instance but the look system won't have it - unless it'd instead be replaced by a system based on the full path (name) of the hierarchies. With USD and materialx having adopted the path based system it makes sense to refactor our system at a point too.

Anyway, I agree - I feel like instances should be allowed - but maybe at least warning or reporting that you will not be able to assign unique shaders?

**Comment by tokejepsen** [comment]
> Anyway, I agree - I feel like instances should be allowed - but maybe at least warning or reporting that you will not be able to assign unique shaders?

Since the `model` family does not support instancing (?) anyways, would it be worth duplicating the meshes and assign cbID when extracting? Essentially baking the instancing to unique meshes.

**Comment by LiborBatek** [comment]
Yeah, we shouldnt convert it to duplicates at all. That beats the purpose of using instances. Typical scenario being model asset with dozens of same geometries acting as instances (for ie piles of dishes in kitchen enviroment). Personally I dont see any problem with not being assing unique shaders to these instances later on when in lookdev. It can be just warning as @BigRoy  proposed already.

In general, instances being used in production for years  we need em working when publishing models. There are maybe workarounds (MASH etc.) but it is essential and basic function in maya so thats why. Also should not break anything in OP besides aformentioned shader assignment limitation.

**Comment by BigRoy** [comment]
I've put together a draft in #4622 - let me know if you could test that and how that works out for you @LiborBatek
